graph/tags.go

```diff
@@ -427,27 +428,32 @@ func validateDigest(dgst string) error {
 	return nil
 }
 
-func (store *TagStore) poolAdd(kind, key string) (chan struct{}, error) {
+// poolAdd checks if a push or pull is already running, and returns (ps, true)
+// if a running operation is found. Otherwise, it creates a new one and returns
+// (ps, false).
+func (store *TagStore) poolAdd(kind, key string) (*progressreader.ProgressStatus, bool) {
```

Comment:
Given we always know if it's a push or pull, it feels odd to have that information passed in an argument. Seems like there should be methods like `pushAdd` and `pullAdd` or something similar.